### DISCUSSION TITLE
Are hooks really disabled by default?

### DISCUSSION THREAD

**Discussion by CViniciusSDias** [swoole/swoole-src#5601] [Q&A] [answered]
Hey there, folks.

[Swoole's documentation](https://wiki.swoole.com/en/#/server/setting?id=hook_flags) states that the `hook_flags` default value is "Do not hook", but I see a different behavior.

This code outputs "Hello, world!", i.e., `sleep` was executed from a hook.

```php
<?php

use function Swoole\Coroutine\{run, go};

run(function () {
    go(function () {
        sleep(2);
        echo 'world!' . PHP_EOL;
    });

    go(function () {
        sleep(1);
        echo 'Hello, ';
    });
});
```

If on the top of the file I add `Co::set(['hook_flags' => 0]);`, then I see the first corouting showing its output first, only then the second one.

PS.: swoole_version() = 5.1.6

**Comment by leocavalcante** [comment, marked as answer]
The `run` function starts a scheduler that enables them: https://github.com/swoole/library/blob/master/src/core/Coroutine/functions.php#L21

**Reply by CViniciusSDias** [reply]
Ah, now I see. I was not even aware of this swoole/library repo. :sweat_smile: 

I'll mark this as the answer, but since we're here, I'd like to bring up a different question...

Using tools such as _Symfony Runtime_, when enabling the hooks shouldn't we see a big difference in benchmarks that have I/O operations such as database queries and/or HTTP calls?
When I'm running simple load tests in a Symfony application running with Swoole with and without `hook_flags` I see no difference.
